.github/workflows/publish_docker_images.yml

```diff
@@ -25,6 +25,15 @@ jobs:
 
       - uses: actions/checkout@v4
 
+      - name: Free up disk space
+        run: |
+          sudo rm -rf /usr/share/dotnet
+          sudo rm -rf /usr/local/lib/android
+          sudo rm -rf /opt/ghc
+          sudo rm -rf /usr/local/share/boost
+          sudo apt clean
+          sudo rm -rf /var/lib/apt/lists/*
```

Comment:
I am not sure that freeing disk space by forcibly removing directories is a good idea. The selected directories look  random. It is unclear whether there are more useless directories. In addition, removing directories directly, rather than uninstalling packages, means that the affected packages are broken. I think this approach is the last resort.

#2183 looks a more maintainable approach.

I suggest excluding the directories deletion logic from this PR and only keeping the improvement of `git clone --depth 1`.